server/src/main/java/org/apache/druid/server/SegmentManager.java

```diff
@@ -265,7 +293,8 @@ public boolean loadSegment(final DataSegment segment, boolean lazy, SegmentLazyL
                 segment.getVersion(),
                 segment.getShardSpec().createChunk(adapter)
             );
-            dataSourceState.addSegment(segment);
+            long numOfRows = lazy ? 0 : adapter.asStorageAdapter().getNumRows();
```

Comment:
We should add a note in the docs that this metric is not reported if lazy loading is enabled.

Also there is a potential NPE here since `adapter.asStorageAdapter()` can return null

Reply:
Can you test this with tombstone segment? (I am not sure if tombstone segment gets load here or not)

Reply:
tested tombstones and made changes to actually count them in the distribution